### ISSUE TITLE
Mypy: Syntax errors

### ISSUE THREAD

**Issue by Dreamsorcerer** [Azure/azure-sdk-for-python#20157]
By mistake, we started using the prerelease version 12.9.0b1 of azure-storage-blob in our CI.

This resulted in these syntax errors from mypy:
```
/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/azure_storage_blob-12.9.0b1-py3.8.egg/azure/storage/blob/_serialize.py:46: error: Type signature has too few arguments [syntax]
/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/azure_storage_blob-12.9.0b1-py3.8.egg/azure/storage/blob/_serialize.py:129: error: Type signature has too few arguments [syntax]
/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/azure_storage_blob-12.9.0b1-py3.8.egg/azure/storage/blob/_blob_client.py:1408: error: Type signature has too few arguments [syntax]
/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/azure_storage_blob-12.9.0b1-py3.8.egg/azure/storage/blob/_blob_client.py:1448: error: Type signature has too few arguments [syntax]
```

If you check the code at those locations, they are obviously wrong as the number of arguments in the type annotation does not match the number of arguments in the definition.

To avoid these mistakes in future, I'd suggest using the common-practice inline annotations, which would make it impossible to make this mistake:
```python
def get_modify_conditions(kwargs: Dict[str, object]) -> ModifiedAccessConditions:
```

**Comment by swathipil** [comment]
@Dreamsorcerer, thanks for the suggestion! Directing it to the right people, so they can address this!

**Comment by amishra-dev** [comment]
Thanks for reporting this @Dreamsorcerer (Sam). We are tracking this work as a part of https://github.com/Azure/azure-sdk-for-python/issues/20083

**Comment by rnestler** [comment]
I fixed some of the type annotations in https://github.com/Azure/azure-sdk-for-python/pull/20084

**Comment by Dreamsorcerer** [comment]
As another suggestion, it would be a good idea to actually setup mypy as a CI check. I can review PRs or answer questions relating to this if help is needed (I've setup several other projects before).

**Comment by dimbleby** [comment]
Broken typing is now released at 12.9.0

**Comment by jalauzon-msft** [comment]
Hi all, sorry for the long delay getting these fixed. I believe these have been addressed in #22741 and #23098 and will be part of our next release of 12.10.
